src/tests/ftest/nvme/nvme_io_verification.py

```diff
@@ -106,3 +106,84 @@ def test_nvme_io_verification(self):
+
+        Use Cases:
+            (1) Running IOR with different set of transfer size where first
+            transfer size is < 4096 and then > 4096. Verify the data after
+            servers are restarted.
+            (2) Repeat the case(1) with maximum nvme pool size that can be
+            created.
+            (3) Running IOR with different set of transfer size where the
+            transfer size is > 4096 throughout. Verify the data after
+            servers are restarted.
+            (4) Repeat the case(3) with maximum nvme pool size that can be
+            created.
+        :avocado: tags=all,full_regression,hw,large,daosio,nvme_server_restart
+        """
+        # Test params
+        tests = self.params.get("ior_sequence", '/run/ior/*')
+        processes = self.params.get("np", '/run/ior/*')
+        transfer_size = self.params.get("tsize", '/run/ior/transfersize/*/')
+        flag_write = self.params.get("write", '/run/ior/*/')
+        flag_read = self.params.get("read", '/run/ior/*/')
+        block_size = self.ior_cmd.block_size.value
+
+        # Loop for every IOR object type
+        for ior_param in tests:
+            # Create and connect to a pool
+            self.pool = TestPool(
+                self.context, dmg_command=self.get_dmg_command())
+            self.pool.get_params(self)
+
+            # update pool sizes
+            self.pool.scm_size.update(ior_param[0])
+            self.pool.nvme_size.update(ior_param[1])
+
+            # Create a pool
+            self.pool.create()
+
+            # get pool info
+            self.pool.get_info()
+
+            for tsize in transfer_size:
+                # Get the current pool sizes
+
+                # Run ior with the parameters specified for this pass
+                self.ior_cmd.transfer_size.update(tsize)
+                self.ior_cmd.flags.update(flag_write)
+                # if transfer size is less thank 1K
+                # update block size to 32K to keep it small
+                if tsize <= 1000:
+                    self.ior_cmd.block_size.update(32000)
+                else:
+                    self.ior_cmd.block_size.update(block_size)
+                self.ior_cmd.set_daos_params(self.server_group, self.pool)
+                self.run_ior(self.get_ior_job_manager_command(), processes)
+
+                # Stop all servers
+                self.get_dmg_command().system_stop(True)
+
+                # Start all servers
+                self.get_dmg_command().system_start()
+
+                # check if all servers started as expected
+                scan_info = self.get_dmg_command().get_output("system_query")
+                if not check_system_query_status(scan_info):
+                    self.fail("One or more servers crashed")
+
+                # read all the data written before server restart
+                self.ior_cmd.flags.update(flag_read)
+                self.run_ior(self.get_ior_job_manager_command(), processes)
```

Comment:
Are we reading the WARNINGS if there is version mismatch because during read even if there is data corruption command will finish as success if I recall correctly.

Reply:
This is already taken care in ior_test_base.py
https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/ior_test_base.py#L242